Content.Shared/Players/ContentPlayerData.cs

```diff
@@ -43,6 +42,31 @@ public sealed class ContentPlayerData
     /// </summary>
     public bool Stealthed { get; set; }
 
+    /// <summary>
+    /// The rate at which this player has been sending messages. Allows for rate-limiting.
+    /// </summary>
```

Comment:
Some of added fields have misleading names. TotalMessageLength and MessageCount are not exactly what they are called - as they are deminished on timer. Maybe call them something like 'LastCheckedMessageRate'? sadly i cant think of a good name right now. Maybe it should be kept as is. At least mention in field comment that it is diminished on timer.